JustSaying.UnitTests/AwsTools/MessageHandling/MessageDispatcherTests/WhenDispatchingMessage.cs

```diff
@@ -68,6 +69,42 @@ protected override MessageDispatcher CreateSystemUnderTest()
             return new MessageDispatcher(_queue, _serialisationRegister, _messageMonitor, _onError, _handlerMap, _loggerFactory, _messageBackoffStrategy);
         }
 
+        public class AndHandlerMapDoesNotHaveMatchingHandler : WhenDispatchingMessage
+        {
+            private const int ExpectedReceiveCount = 1;
+            private readonly TimeSpan _expectedBackoffTimeSpan = TimeSpan.FromMinutes(4);
+
+            protected override void Given()
+            {
+                base.Given();
+                _messageBackoffStrategy.GetBackoffDuration(_typedMessage, 1, null).Returns(_expectedBackoffTimeSpan);
+                _sqsMessage.Attributes.Add(MessageSystemAttributeName.ApproximateReceiveCount, ExpectedReceiveCount.ToString());
+            }
+
+            [Fact]
+            public void ShouldDeserializeMessage()
+            {
+                _serialisationRegister.Received(1).DeserializeMessage(Arg.Is<string>(x => x == _sqsMessage.Body));
+            }
+
+            [Fact]
+            public void ShouldLogError()
+            {
+                _logger.Received().Log(
```

Comment:
I don't think we should be asserting logging at this level, because it's brittle and breaks tests with changes that shouldn't break tests.

It should be enough that the handler returns false and that we update the visibility, _maybe_ that _an_ error is logged, but not the whole log message.

Reply:
Turns out this is not even possible in master: https://github.com/justeat/JustSaying/pull/728

What do others think?

Reply:
Agreed, the log message is a side effect, not really something that should be asserted on IMO

Reply:
Done.